crates/cubecl-core/src/ir/kernel.rs

```diff
@@ -171,7 +171,8 @@ impl Elem {
                 UIntKind::U32 => core::mem::size_of::<u32>(),
                 UIntKind::U64 => core::mem::size_of::<u64>(),
             },
-            Elem::Bool => core::mem::size_of::<bool>(),
+            // Currently, bools are represented as u32 in the backend.
+            Elem::Bool => core::mem::size_of::<u32>(),
```

Comment:
This isn't true! Bools are still boleans in the kernel, we just can't load and save them to global memory. That's why we're using `u32` in Burn to represent BoolTensor, but when fusing them, they never actually materialize as u32.